### ISSUE TITLE
label[] breaks multiline translations

### ISSUE THREAD

**Issue by MineWitherMC** [minetest/minetest#7450]
##### Issue type
 - Bug report
 
##### Minetest version
```
minetest-0.5.0-dev-9a1d3584
```

##### Summary
`label[x,y;text]` element breaks `minetest.get_translation` when the latter has line breaks.

##### Steps to reproduce
Write a following mod:
```lua
local S = minetest.get_translator("test")

minetest.register_on_joinplayer(function(player)
	minetest.show_formspec(player:get_player_name(), "test", ([[
		size[4,4]
		label[0,0;%s]
		textarea[0,2;4,2;;;%s]
	]]):format(S"A@nB@nC", S"A@nB@nC"))
end)
```

#### Expected Behavior
Both strings are translated into the same one or give a single warning.

#### Actual Behavior
If translation for `A@nB@nC` is present, only textarea is translated. I get 2 warnings about this:
```
2018-06-15 10:24:31: WARNING[Main]: Translations: can't find translation for string "A" in textdomain "test"
2018-06-15 10:24:31: WARNING[Main]: Translations: can't find translation for string "A
2018-06-15 10:24:31: WARNING[Main]: B
2018-06-15 10:24:31: WARNING[Main]: C" in textdomain "test"
```
First one is because of Label, second is because of Textarea.

**Comment by sfan5** [comment]
does `formspec_escape` help here?

**Comment by MineWitherMC** [comment]
No.

**Comment by SmallJoker** [comment]
Looks more like an issue with translations instead of the formspec code. Does `S("A@nB@nC")` work alone? - and does it produce the correct output?

**Comment by MineWitherMC** [comment]
Yeah. The problem is that it works perfectly well with item names, textareas, etc. - but not with labels.

**Comment by SmallJoker** [comment]
The issue with `label[]` here is that each line has its own `gui::StaticText` element. This means that also coloring multiple lines is broken for multiline labels.

**Comment by doxygen-spammer** [comment]
Here exactly is the bug, in GUIFormSpecMenu::parseLabel():

https://github.com/minetest/minetest/blob/a463620edbe57071a7101297d33226507567ca73/src/gui/guiFormSpecMenu.cpp#L1737-L1748

In line 1737, you see the not-yet-translated string being broken at newline characters.
In line 1745, you see the attempt to translate the lines individually.
Of course, none of these lines has any valid translation markup left.

I tried to fix this locally.
It is just to move the translation from the for loop to before the split() call.
The translation involves conversion from std::string to std::wstring, but split() can handle that.
Only the separator needs to be changed from `'\n'` to `L'\n'`.

GUIFormSpecMenu::parseVertLabel() is located directly below, I copied the translation code from there.
But I do not know the details of unescape_string(), unescape_translate(), unescape_enriched(), and which of them is correct.

To my understanding, the horizontal label is the only affected formspec element, because no other element splits text by lines (or anything).
